triage/weekly_report.py

```diff
@@ -172,6 +174,17 @@ def gh_link(pr):
     return f'https://github.com/rust-lang/rust/issues/{pr}'
 
 
+def gh_pr_title(pr):
+    if pr in pr_titles:
+        return pr_titles.get(pr)
+
+    url = f'https://api.github.com/repos/rust-lang/rust/pulls/{pr}'
```

Comment:
I think unfortunately this is going to be pretty error prone -- we land 7-10 PRs/day (without rollups, which wouldn't be noted here regardless). That means that over a 7 day period we'd have roughly ~50-55 PRs, and GitHub's rate limit without authentication is 60 requests per hour. Even running the script twice wouldn't be possible. Would you be up for adding authentication here? It should be mostly a matter of setting the "Authorization" header to something like "token $GITHUB_TOKEN", which is optionally present in the environment.

Reply:
@Mark-Simulacrum, you are right, will fix it;
- besides limit 60 req/h; is msg format ok? "Fix SGX CI, take 3 #79797" `'{title} [#{pr}]({gh_link(pr)})'`

Reply:
Yes, that seems fine, thanks!